common/unsafe/src/main/java/org/apache/spark/unsafe/UnsafeAlignedOffset.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Class to make changes to record length offsets uniform through out
+ * various areas of Apache Spark core and unsafe.  The SPARC platform
+ * requires this because using a 4 byte Int for record lengths causes
+ * the entire record of 8 byte Items to become misaligned by 4 bytes.
+ * Using a 8 byte long for record length keeps things 8 byte aligned.
+ */
+public class UnsafeAlignedOffset {
+
+  private static final int UAO_SIZE;
+
+  static {
+    if (Arrays.asList("sparc", "sparcv9").contains(System.getProperty("os.arch"))) {
```

Comment:
OK, this is looking strong, because it doesn't now affect the current behavior for x86 et al. There are three places where this test above is implemented. It could be exposed from here as a static method and that can be used to assign ByteArrayMethods.alignedArch. I'm not sure about CompressibleColumnBuilder but can it be accessed there too? I think so.